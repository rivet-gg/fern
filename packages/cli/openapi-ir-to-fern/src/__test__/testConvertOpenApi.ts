import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { parse } from "@fern-api/openapi-parser";
import { createMockTaskContext } from "@fern-api/task-context";
import path from "path";
import { convert } from "../convert";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testConvertOpenAPI(fixtureName: string, filename: string, asyncApiFilename?: string): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("simple", async () => {
            const openApiPath = path.join(FIXTURES_PATH, fixtureName, filename);
            const mockTaskContext = createMockTaskContext({ logger: CONSOLE_LOGGER });

            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;

            const specs = [];
            specs.push({
                absoluteFilepath: AbsoluteFilePath.of(openApiPath),
                absoluteFilepathToOverrides: undefined
            });
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const fernDefinition = convert({
                ir: openApiIr,
                taskContext: mockTaskContext,
                enableUniqueErrorsPerEndpoint: false,
                detectGlobalHeaders: true
            });
            expect(fernDefinition).toMatchSnapshot();
        });

        it("docs", async () => {
            const openApiPath = path.join(FIXTURES_PATH, fixtureName, filename);
            const mockTaskContext = createMockTaskContext({ logger: CONSOLE_LOGGER });

            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;

            const specs = [];
            specs.push({
                absoluteFilepath: AbsoluteFilePath.of(openApiPath),
                absoluteFilepathToOverrides: undefined
            });
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const fernDefinition = convert({
                ir: openApiIr,
                taskContext: mockTaskContext,
                enableUniqueErrorsPerEndpoint: true,
                detectGlobalHeaders: false
            });
            expect(fernDefinition).toMatchSnapshot();
        });
    });
}
