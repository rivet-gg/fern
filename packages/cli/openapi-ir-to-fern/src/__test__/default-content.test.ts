import { testConvertOpenAPI } from "./testConvertOpenApi";

describe("open api ir to fern", () => {
    testConvertOpenAPI("default-content", "openapi.yml");
});
