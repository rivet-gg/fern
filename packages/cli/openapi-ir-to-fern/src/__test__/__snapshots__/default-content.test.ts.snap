// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open api ir to fern default-content docs 1`] = `
{
  "definitionFiles": {
    "generativelanguage.yml": {
      "imports": {
        "root": "__package__.yml",
      },
      "service": {
        "auth": false,
        "base-path": "",
        "endpoints": {
          "GenerateContent": {
            "auth": false,
            "docs": undefined,
            "examples": [
              {
                "path-parameters": {
                  "model": "model",
                },
                "response": {
                  "body": {
                    "bar": "bar",
                  },
                },
              },
            ],
            "method": "POST",
            "pagination": undefined,
            "path": "/generateContent",
            "path-parameters": {
              "model": "string",
            },
            "response": {
              "docs": "Successful operation",
              "type": "root.GenerateContentResponse",
            },
          },
        },
      },
    },
  },
  "packageMarkerFile": {
    "types": {
      "GenerateContentResponse": {
        "docs": undefined,
        "properties": {
          "bar": "optional<string>",
        },
      },
    },
  },
  "rootApiFile": {
    "default-environment": "Default",
    "display-name": "Default Content Type",
    "environments": {
      "Default": "https://ai.com",
    },
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;

exports[`open api ir to fern default-content simple 1`] = `
{
  "definitionFiles": {
    "generativelanguage.yml": {
      "imports": {
        "root": "__package__.yml",
      },
      "service": {
        "auth": false,
        "base-path": "",
        "endpoints": {
          "GenerateContent": {
            "auth": false,
            "docs": undefined,
            "examples": [
              {
                "path-parameters": {
                  "model": "model",
                },
                "response": {
                  "body": {
                    "bar": "bar",
                  },
                },
              },
            ],
            "method": "POST",
            "pagination": undefined,
            "path": "/generateContent",
            "path-parameters": {
              "model": "string",
            },
            "response": {
              "docs": "Successful operation",
              "type": "root.GenerateContentResponse",
            },
          },
        },
      },
    },
  },
  "packageMarkerFile": {
    "types": {
      "GenerateContentResponse": {
        "docs": undefined,
        "properties": {
          "bar": "optional<string>",
        },
      },
    },
  },
  "rootApiFile": {
    "default-environment": "Default",
    "display-name": "Default Content Type",
    "environments": {
      "Default": "https://ai.com",
    },
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;
